frontend/components/Fields/BranchField.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Field } from 'redux-form';
+import RenderUrlField from './RenderUrlField';
+
+const invalidPrefix = /^[/|.]/;
+const invalidChars = /(.*\.{2}|.*\/{2}|^@$|.*@\{|[\s<>~^:?[\\])/;
+const invalidSuffix = /(\.lock|\.|\/)$/;
+
+const validations = [invalidPrefix, invalidChars, invalidSuffix];
+
+const validate = string =>
+  validations.some(validation => validation.test(string));
+
+export const validateBranchName = (value) => {
+  if (value && value.length && !!validate(value)) {
+    return 'Branch name contains invalid characters.';
+  }
+
+  return undefined;
+};
+
+const BranchField = ({ ...props }) =>
+  <Field
+    component={RenderUrlField}
```

Comment:
I think this is the wrong component to use here since `RenderUrlField` adds `type="url"` to the underlying `input` element.

My intention (though not specified anywhere) was that we'd have different `RenderXYZField` components that could be shared among similar custom Field components. Like `RenderUrlField` is used in both `HttpsUrlField` and `GitHubRepoUrlField`.

So, maybe you could either make a new `RenderTextField` component that has `type="text"` and other underlying attributes set appropriately, OR further abstract `RenderUrlField` to something like `RenderInputField` and move the `type` and `autoComplete` up a level.

Hopefully this all makes sense. I'm not that well-versed in redux-form and it has a somewhat bizarre API.

Reply:
I think the best strategy is actually to rename this to `FieldWithError` or something similarly descriptive, since that is what it's actual behavior seems to be.

The `type` of the field can just be controlled with props, and since we need to wrap these fields in an additional component to handle validations, I don't think we need to necessarily have multiple components that differ only by their `type`

Reply:
Yeah, that sounds good to me!